### ISSUE TITLE
Additional properties generated in a header ref

### ISSUE THREAD

**Issue by michaeljmcd** [smallrye/smallrye-open-api#274]
When I include a ref in a header, like the one below:

      headers = { @Header(ref = "my-header"), },

The resulting YAML generation looks like:

            my-header:
              $ref: my-header
              style: simple

The extra `style: simple` property then causes validation with Spectral to fail, with the error:

           error  no-$ref-siblings             $ref cannot be placed next to any other properties

Which at least broadly aligns with the OpenAPI spec (https://swagger.io/specification/#referenceObject) which says that "This object cannot be extended with additional properties and any properties added SHALL be ignored."

Looking through the code, I think the root cause is that `HeaderImpl.java` defaults the style to `SIMPLE` and `OpenApiSerializer.writeHeader()` doesn't exclude properties when `$ref` is set. It looks like adding a bit of logic to the serializer to skip over other properties when a ref is present would fix the issue. I'd be happy to open a PR if there is some agreement that there shouldn't be any siblings to `$ref`.

**Comment by MikeEdgar** [comment]
+1

The JavaDoc for the `@Header` annotation's `ref` property states that if it is used in addition to other properties, the result is undefined. My opinion is that if the serializer encounters a `ref` as well as other properties, only the `ref` should be serialized. I think that works within the bounds of the specification. 

There are TCK tests that check for the default `simple` value, so I think that default needs to remain in place in `HeaderImpl`.

```java
    /**
     * Reference value to a Header object.
     * <p>
     * This property provides a reference to an object defined elsewhere. This property and
     * all other properties are mutually exclusive. If other properties are defined in addition
     * to the ref property then the result is undefined.
     * 
     * @return reference to a header
     **/
    String ref() default "";
```

**Comment by michaeljmcd** [comment]
That's a good callout on the text in the javadoc description. A quick `grep` shows it to exist in MicroProfile OpenAPI in these classes:

    api/src/main/java/org/eclipse/microprofile/openapi/annotations/callbacks/Callback.java
    api/src/main/java/org/eclipse/microprofile/openapi/annotations/headers/Header.java
    api/src/main/java/org/eclipse/microprofile/openapi/annotations/links/Link.java
    api/src/main/java/org/eclipse/microprofile/openapi/annotations/media/ExampleObject.java
    api/src/main/java/org/eclipse/microprofile/openapi/annotations/media/Schema.java
    api/src/main/java/org/eclipse/microprofile/openapi/annotations/parameters/Parameter.java
    api/src/main/java/org/eclipse/microprofile/openapi/annotations/parameters/RequestBody.java
    api/src/main/java/org/eclipse/microprofile/openapi/annotations/media/SchemaProperty.java
    api/src/main/java/org/eclipse/microprofile/openapi/annotations/responses/APIResponse.java
    api/src/main/java/org/eclipse/microprofile/openapi/annotations/security/SecurityScheme.java
    api/src/main/java/org/eclipse/microprofile/openapi/annotations/tags/Tag.java

I'll use that as a starting point for a branch.

**Comment by michaeljmcd** [comment]
I've opened PR #286 with the first couple of entities. If there is still interest and these changes look good, I'd be glad to finish it out and clean up the merge status.

**Comment by MikeEdgar** [comment]
Fixed by #286
